https://phabricator.kde.org/D7871

It was possible that we performed an early exit when the caption changes
and this results in the visible name being wrong. E.g if we have two
windows called foo, the second one has the visible name "foo <2>". After
changing that to "bar" the bug resulted in it still being the foo
variant instead of getting cleared.

BUG: 384760
FIXED-IN: 5.11.0

diff --git client.cpp client.cpp
--- client.cpp
+++ client.cpp
@@ -1447,8 +1447,9 @@
     for (int i = 0; i < s.length(); ++i)
         if (!s[i].isPrint())
             s[i] = QChar(u' ');
+    const bool changed = (s != cap_normal);
     cap_normal = s;
-    if (!force) {
+    if (!force && !changed) {
         emit captionChanged();
         return;
     }
